src/screen/splash/mod.rs

```diff
@@ -2,26 +2,28 @@
 
 use bevy::{asset::embedded_asset, prelude::*};
 
-use crate::util::prelude::*;
-
 use super::Screen;
+use crate::util::prelude::*;
 
 pub(super) fn plugin(app: &mut App) {
     // Add splash image
     embedded_asset!(app, "splash.png");
-    app.add_systems(OnEnter(Screen::Splash), spawn_splash)
-        // Make the screen-flicker bug less jarring on Windows (see `core/deflicker.rs`).
-        .insert_resource(ClearColor(SPLASH_BACKGROUND_COLOR));
+    app.add_systems(OnEnter(Screen::Splash), spawn_splash);
+    // Make the screen-flicker bug less jarring on Windows
+    // (see `core/deflicker.rs`).
+    app.insert_resource(ClearColor(SPLASH_BACKGROUND_COLOR));
 
     // Add splash timer
-    app.add_systems(OnEnter(Screen::Splash), insert_splash_timer)
-        .add_systems(OnExit(Screen::Splash), remove_splash_timer)
-        .add_systems(
-            Update,
-            (tick_splash_timer, check_splash_timer)
-                .chain()
-                .run_if(in_state(Screen::Splash)),
-        );
+    app.add_systems(OnEnter(Screen::Splash), insert_splash_timer);
+    app.add_systems(OnExit(Screen::Splash), remove_splash_timer);
+    app.add_systems(
+        Update,
+        (tick_splash_timer, check_splash_timer)
+            .chain()
+            .run_if(in_state(Screen::Splash)),
+    );
+
+    app.register_type::<SplashTimer>();
```

Comment:
```suggestion
    app.register_type::<SplashTimer>();
    app.add_systems(OnEnter(Screen::Splash), insert_splash_timer);
    app.add_systems(OnExit(Screen::Splash), remove_splash_timer);
    app.add_systems(
        Update,
        (tick_splash_timer, check_splash_timer)
            .chain()
            .run_if(in_state(Screen::Splash)),
    );
```

Grouping this up with the other splash timer logic.